script.R

```diff
@@ -18,9 +65,18 @@ timeout <- as.integer(args[3])
 
 # Install required packages
 catnl("Installing required packages...")
-install.packages(c(
-  "pak"
-))
+if (!requireNamespace("pak", quietly = TRUE)) {
+  install.packages("pak", quiet = TRUE)
+}
+if (!requireNamespace("pkgcache", quietly = TRUE)) {
+  install.packages("pkgcache", quiet = TRUE)
+}
+options(
+  repos = c(
+    PPM = pkgcache::repo_resolve("PPM@latest"),
+    getOption("repos")
+  )
+)
```

Comment:
In the next list we have pkgs installation 

https://github.com/insightsengineering/r-revdepcheck-action/blob/1fc867c2e149a60fb0de86da26ac5c68516a9c14/script.R#L80-L88

what if those were already installed? Should we set diff with `rownames(installed.packages())`?

Reply:
This is clever enough to install only missing. No need to handle this on our side